src/Notes/WC_Admin_Notes_Tracking_Opt_In.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * WooCommerce Admin Usage Tracking Opt In Note Provider.
+ *
+ * Adds a note to the merchant's inbox showing the benefits of the Facebook extension.
+ *
+ * @package WooCommerce Admin
+ */
+
+namespace Automattic\WooCommerce\Admin\Notes;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Notes_Tracking_Opt_In
+ */
+class WC_Admin_Notes_Tracking_Opt_In {
+	/**
+	 * Note traits.
+	 */
+	use NoteTraits;
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-admin-usage-tracking-opt-in';
+
+	/**
+	 * Attach hooks.
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_admin_note_action_tracking-opt-in', array( $this, 'opt_in_to_tracking' ) );
+	}
+
+	/**
+	 * Possibly add Usage Tracking Opt In extension note.
+	 */
+	public static function possibly_add_tracking_opt_in_note() {
+		// Only show this note to stores that are opted out.
+		if ( 'yes' === get_option( 'woocommerce_allow_tracking', 'no' ) ) {
+			return;
+		}
+
+		// We want to show the note after one week.
+		if ( ! self::wc_admin_active_for( WEEK_IN_SECONDS ) ) {
+			return;
+		}
+
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+
+		// We already have this note? Then exit, we're done.
+		$note_ids = $data_store->get_notes_with_name( self::NOTE_NAME );
+		if ( ! empty( $note_ids ) ) {
+			return;
+		}
+		
+		/* translators: 1: link to WooCommerce.com settings, 2: link to WooCommerce.com tracking documentation. */
+		$content_format = __(
+			'Gathering usage data allows us to improve WooCommerce. Your store will be considered as we evaluate new features, judge the quality of an update, or determine if an improvement makes sense. You can always visit the <a href="%1$s" target="_blank">Settings</a> and choose to stop sharing data. <a href="%2$s" target="_blank">Read more</a> about what data we collect.',
```

Comment:
HTML should not be passed to translation functions. Once the markup has been moved to parameters then this could be passed through `esc_html__` instead of `wp_kses`.

Reply:
Can you elaborate on why we shouldn't pass limited markup in this case?

* The Plugin Developers Handbook [shows an example](https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#descriptions) with markup
* Core WordPress passes markup to translation functions [in several places](https://github.com/WordPress/WordPress/search?q=__%28+wp_kses&unscoped_q=__%28+wp_kses)
* WooCommerce Admin's [use of `interpolateComponents`](https://github.com/woocommerce/woocommerce-admin/blob/c7b35aaa1598d7bd4663ad131f7eb161f37c4f84/client/dashboard/profile-wizard/steps/usage-modal.js#L66-L76) isn't much different

Reply:
It's listed in the best practices as something to avoid - https://developer.wordpress.org/themes/functionality/internationalization/#best-practices-for-writing-strings

Reply:
Perhaps we need a tie breaker. The theme handbook says "Do not put unnecessary HTML markup into the translated string".

Including the links with their text allows the string to be translated with the full context. Removing the links would in turn create 3 strings to be translated:

1. `'Gathering usage data allows us to improve WooCommerce. Your store will be considered as we evaluate new features, judge the quality of an update, or determine if an improvement makes sense. You can always visit the %1$s and choose to stop sharing data. %2$s about what data we collect.'`
1. `'Settings'`
1. `'Read more'`

_cc: @psealock_

Reply:
I was suggesting

```
/* translators: 1: open link to WooCommerce.com settings, 2: open link to WooCommerce.com tracking documentation, 3: closing link tag. */
		$content_format = __(
			'Gathering usage data allows us to improve WooCommerce. Your store will be considered as we evaluate new features, judge the quality of an update, or determine if an improvement makes sense. You can always visit the %1$sSettings%$3s and choose to stop sharing data. %2$sRead more%$3s about what data we collect.'
```

That leaves just one string to translate by including the link markup in the placeholder substitutions.

Reply:
Updated in cc39870.